rpc/json/types.go

```diff
@@ -160,6 +161,54 @@ func (s *StrInt) UnmarshalJSON(b []byte) error {
 	return err
 }
 
+// BlockNumber is a StrInt64 with helper tags for block heights
+type BlockNumber StrInt64
+
+// BlockNumber tags:
+// - "earliest" = literal 1
+// - "included" = seen on DA
+const (
+	IncludedBlockNumber = BlockNumber(-1)
+	EarliestBlockNumber = BlockNumber(1)
+)
+
+func unmarshalBlockNumber(b []byte, bn *BlockNumber) error {
```

Comment:
i don't think you need a separate function. just move the code to `UnmarshalJSON` and in tests call the function over the `BlockNumber` instance.

Reply:
Fixed.